compiler/rustc_codegen_llvm/src/context.rs

```diff
@@ -290,6 +290,11 @@ pub(crate) unsafe fn create_module<'ll>(
         );
     }
 
+    // Disable ThinLTO if fat lto is requested. Otherwise lld defaults to thin lto.
+    if sess.lto() == config::Lto::Fat {
+        llvm::add_module_flag_u32(llmod, llvm::ModuleFlagMergeBehavior::Override, "ThinLTO", 0);
+    }
```

Comment:
What if a dependency is built with lto=true (aka lto=fat), but then the user wants to use thinLTO? I'm pretty sure the standard library is built with lto=true for example, but that shouldn't prevent thinLTO from ever working.

Reply:
Good question, it seems to change somewhat, but still work in general. I added a test for this.
What changes: Without this change, the test passes when
lib is compiled with `O0` and main with `O3` and
1. lib uses lto=thin and main uses lto=thin
2. lib uses lto=thin and main uses lto=fat
3. lib uses lto=fat and main uses lto=thin
4. lib uses lto=fat and main uses lto=fat

With this change, all of these keep passing except for case 3 (lib uses lto=fat and main uses lto=thin).
When lib is compiled with `O1`, `O2` or `O3`, case 3 passes as well.
I assume this is the important case, as the standard library is compiled with optimizations.
(And lto with `O0` is kinda questionable, except maybe for nvptx and amdgpu, but they require lto=fat anyway.)